tests/lib/rules/no-array-prototype-extensions.js

```diff
@@ -233,10 +233,33 @@ ruleTester.run('no-array-prototype-extensions', rule, {
       output: null,
       errors: [{ messageId: 'main', type: 'CallExpression' }],
     },
+    {
+      code: `
+      const arr = [];
+
+      function getAge() {
+        return 16;
+      }
+
+      arr.filterBy("age", getAge());
+      `,
+      output: `
+      import { get } from '@ember/object';
+const arr = [];
+
+      function getAge() {
+        return 16;
+      }
+
+      arr.filter(item => get(item, "age") === getAge());
+      `,
+      errors: [{ messageId: 'main', type: 'CallExpression' }],
+    },
     {
       // Set in variable name but not a Set function.
-      code: 'set.filterBy()',
-      output: null,
+      code: 'set.filterBy("age", 18);',
+      output: `import { get } from '@ember/object';
+set.filter(item => get(item, "age") === 18);`,
```

Comment:
filterBy can accept the value as optional value. Could we add another test for `set.filterBy("age")` (no value) also? https://api.emberjs.com/ember/release/classes/EmberArray/methods/mapBy?anchor=filterBy

Reply:
Good callout. Didn't know that the second argument is optional. Will fix it.